habitat/tasks/nav/nav.py

```diff
@@ -1015,6 +1015,46 @@ def update_metric(
             self._metric = distance_to_target
 
 
+@registry.register_measure
+class DistanceToGoalReward(Measure):
+    """
+    The measure calculates a reward based on the distance towards the goal.
+    The reward is simply - (new_distance - previous_distance) i.e. the
+    decrease of distance to the goal.
+    """
+
+    cls_uuid: str = "distance_to_goal_reward"
+
+    def __init__(
+        self, sim: Simulator, config: Config, *args: Any, **kwargs: Any
+    ):
+        self._sim = sim
+        self._config = config
+        self._previous_distance: Optional[float] = None
+        super().__init__()
+
+    def _get_uuid(self, *args: Any, **kwargs: Any) -> str:
```

Comment:
@vincentpierre BTW, a longer term thing transicriont I forgot to mention is deprecating these entirely as these UUIDs can be defined from the YAML if they exist in the sensor config. I don't recall if that's the case with measures, but we may want to make ti so.

Reply:
This is a good idea to replace, but I think this should be a separate PR since so many sensors / measures define the UUID in the class as opposed to YAML.

Reply:
I will write it down in the doc